packages/convex-helpers/server/index.ts

```diff
@@ -68,3 +79,105 @@ export function deploymentName() {
+ * You can expose these operations in your API. For example, in convex/users.ts:
+ *
+ * ```ts
+ * // in convex/users.ts
+ * import { crud } from "convex-helpers/server";
+ * import { query, mutation } from "./convex/_generated/server";
+ * const Users = Table("users", {
+ *  name: v.string(),
+ *  ///...
+ * });
+ * export const { Create, Read, Update, Delete } = crud(Users, query, mutation);
+ * ```
+ *
+ * Then from a client, you can access `api.users.create`.
+ *
+ * @param table The table to create CRUD operations for.
+ * Of type returned from Table() in "convex-helpers/server".
+ * @param query The query to use - use internalQuery or query from
+ * "./convex/_generated/server" or a customQuery.
+ * @param mutation The mutation to use - use internalMutation or mutation from
+ * "./convex/_generated/server" or a customMutation.
+ * @returns An object with create, read, update, and delete functions.
+ */
+export function crud<
+  Fields extends Record<string, Validator<any, any, any>>,
+  TableName extends string,
+  DataModel extends GenericDataModel,
+  QueryVisibility extends FunctionVisibility,
+  MutationVisibility extends FunctionVisibility
+>(
+  table: {
+    name: TableName;
+    _id: Validator<GenericId<TableName>>;
+    withoutSystemFields: Fields;
+  },
+  query: QueryBuilder<DataModel, QueryVisibility>,
+  mutation: MutationBuilder<DataModel, MutationVisibility>
+) {
+  return {
+    Create: mutation({
+      args: table.withoutSystemFields,
+      handler: async (ctx, args) => {
+        return await ctx.db.insert(
+          table.name,
+          args as unknown as WithoutSystemFields<
+            DocumentByName<DataModel, TableName>
+          >
+        );
+      },
+    }) as RegisteredMutation<
+      MutationVisibility,
+      ObjectType<Fields>,
+      Promise<GenericId<TableName>>
+    >,
+    Read: query({
+      args: { id: table._id },
+      handler: async (ctx, args) => {
+        return await ctx.db.get(args.id);
+      },
+    }) as RegisteredQuery<
+      QueryVisibility,
+      { id: GenericId<TableName> },
+      Promise<DocumentByName<DataModel, TableName> | null>
+    >,
+    Update: mutation({
+      args: {
+        id: v.id(table.name),
+        patch: v.object(partial(table.withoutSystemFields)),
+      },
+      handler: async (ctx, args) => {
+        await ctx.db.patch(
+          args.id,
+          args.patch as Partial<DocumentByName<DataModel, TableName>>
+        );
+      },
+    }) as RegisteredMutation<
+      MutationVisibility,
+      {
+        id: GenericId<TableName>;
+        patch: Partial<
+          WithoutSystemFields<DocumentByName<DataModel, TableName>>
+        >;
+      },
+      Promise<void>
+    >,
+    Delete: mutation({
+      args: { id: table._id },
+      handler: async (ctx, args) => {
+        await ctx.db.delete(args.id);
+      },
+    }) as RegisteredMutation<
+      MutationVisibility,
+      { id: GenericId<TableName> },
+      Promise<void>
+    >,
+  };
```

Comment:
I made them all title case b/c I couldn't define `delete`: I considered `delete_` but it didn't feel right. and `remove` doesn't match the D in CRUD

Reply:
In templates-cms I used "destroy" for this, just an idea in case you want to stick to lowercase as more in keeping with the rest of examples etc - but I also think title case is OK

Reply:
oh I just saw the later commit going back to "delete_", I think personally I'd prefer sticking to consistent naming across all 4 - either all title case or "destroy"